src/pages/api/dogs/[id].js

```diff
@@ -73,7 +73,7 @@ export default async function handler(req, res) {
         !data.caregivers?.some((o) => o._id.equals(user._id))
       ) {
         if (
-          data.partner?.user?.equals(user._id) ||
+          data.partner?.email?.equals(user.email) ||
```

Comment:
should just use data.partner?.email === user.email since both are just strings. I get this error on GET /api/dogs/id requests:
<img width="550" alt="Screenshot 2024-04-13 at 7 37 04 PM" src="https://github.com/GTBitsOfGood/canine-assistants/assets/45372449/c5059a67-a0cb-451d-aaeb-7580c7e89bf7">
<img width="575" alt="Screenshot 2024-04-13 at 7 36 49 PM" src="https://github.com/GTBitsOfGood/canine-assistants/assets/45372449/1614a998-c027-43a3-9a73-b8587be0a9fb">